tomcat-gcp-lib/src/test/java/com/google/cloud/runtimes/tomcat/logging/TraceValveTest.java

```diff
@@ -0,0 +1,101 @@
+package com.google.cloud.runtimes.tomcat.logging;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.trace.SpanContextHandler;
+import com.google.cloud.trace.Tracer;
+import com.google.cloud.trace.core.ConstantTraceOptionsFactory;
+import com.google.cloud.trace.core.SpanContext;
+import com.google.cloud.trace.core.SpanContextFactory;
+import com.google.cloud.trace.core.TraceContext;
+import com.google.cloud.trace.service.TraceService;
+import com.google.devtools.cloudtrace.v1.TraceSpan;
+import org.apache.catalina.Valve;
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
+import org.apache.tomcat.util.descriptor.web.ContextHandler;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class TraceValveTest {
+
+  @Mock
+  private TraceService traceService;
+
+  @Mock
+  private Tracer tracer;
+
+  @Mock
+  private Request request;
+
+  @Mock
+  private Response response;
+
+  @Mock
+  private Valve nextValve;
+
+  @Mock
+  private TraceContext traceContext;
+
+  @Mock
+  private SpanContextHandler spanContextHandler;
+
+  @Mock
+  private SpanContextFactory spanContextFactory;
+
+  @InjectMocks
+  private TraceValve valve;
+
```

Comment:
I think it would also be good to test the behavior when the TraceValve is not configured correctly (i.e. if traceDelay == 0).